File: twoSum.py
def twoSum(numbers: list[int], target: int) -> list[int]:
    left, right = 0, len(numbers) - 1
    while left < right:
        temp_sum = numbers[left] + numbers[right]

        if temp_sum == target:
            return [left, right]
        elif temp_sum > target:
            right -= 1
        else:
            left += 1


l = [2, 7, 11, 15]
target = 9

print(twoSum(l, target))

File: twoSum.go
package main

import "fmt"

func main() {
	nums := []int{5, 25, 75}
	target := 100
	fmt.Printf("twoSum(nums, target): %v\n", twoSum(nums, target))
}

func twoSum(numbers []int, target int) []int {
	left, right := 0, len(numbers)-1
	for left < right {
		tempSum := numbers[left] + numbers[right]
		if tempSum == target {
			return []int{left + 1, right + 1}
		} else if tempSum > target {
			right--
		} else {
			left++
		}
	}
	return []int{-1}
}
